Add statsD receiver documentation (#76)
aws-otel/aws-otel.github.io: src/docs/components/statsd-receiver.mdx

---
title: 'Getting Started with StatsD Receiver in AWS OpenTelemetry Collector'
description:
    StatsD receiver is an agent that collects [StatsD metrics](https://github.com/statsd/statsd/blob/master/docs/metric_types.md)
    and do the aggregation in a customer defined interval(default value is 60s). You can send counters and gauges to StatsD receiver.
    The StatsD receiver will send the aggregated metrics to the following workflow.
path: '/docs/components/statsd-receiver'
---

import SectionSeparator from "components/MdxSectionSeparator/sectionSeparator.jsx"

StatsD receiver is an agent that collects [StatsD metrics](https://github.com/statsd/statsd/blob/master/docs/metric_types.md)

> and do the [aggregation](#aggregation) during a customer defined interval (default value is 60s). The aggregation interval is similar to the flush

Reviewer: ```suggestion
and does [aggregation](#aggregation) for a defined interval (default value is 60s). The aggregation interval is similar to the flush
```

> interval in StatsD. You can send counters and gauges to StatsD receiver. The StatsD receiver will send the aggregated metrics to the following workflow.

Reviewer: ```suggestion
interval in StatsD. You can send counters and gauges to StatsD receiver, which will send the aggregated metrics to the following workflow.
```

For example, you can use EMF exporter to send the metrics to CloudWatch.

> StatsD receiver now supports counter and gauge types which covers most of use cases. It listens to a port which can be

Reviewer: ```suggestion
StatsD receiver currently supports counter and gauge types. The StatsD receiver can be used as a replacement
for the CloudWatch agent if these are the only types you use.
```

treated as a replacement of CloudWatch agent for StatsD.

<SectionSeparator />

## Enabling StatsD Receiver

> To enable the StatsD receiver, add the name under receiver section in the config file `(local/config.yaml)`.

Reviewer: ```suggestion
To enable the StatsD receiver, add the name under the receiver section in the config file `(local/config.yaml)`.
```

By default, the receiver listens address `localhost:8125`, collects all metrics and aggregate the metrics in 60s.
The `transport` can only be udp.
(For the full list of metrics, see [Available Metrics](#available-metrics)).

The following configuration collects StatsD metrics by using StatsD receiver and sends

> them to CloudWatch using `awsemf` exporter. You could change the port name, aggregation_interval et.al to the values
> suits you the most. Check out [SETUP](https://aws-otel.github.io/docs/setup/ecs) section for

Reviewer: ```suggestion
them to CloudWatch using `awsemf` exporter. You can set these configuration values to what works
for your application. Check out [SETUP](https://aws-otel.github.io/docs/setup/ecs) section for
```

configuring AWS Distro for OpenTelemetry Collector in Amazon Elastic Container Service.

```yaml lineNumbers=true
receivers:
  statsd:
    endpoint: 0.0.0.0:8125
    aggregation_interval: 60s
exporters:
  awsemf:
    namespace: ECS/AWSOTel/Application
    log_group_name: '/aws/ecs/application/metrics'
    region: 'us-west-2'
service:
  pipelines:
    metrics:
      receivers: [statsd]
      exporters: [awsemf]
```

<SectionSeparator />

## Available Metrics

### General format:

`<name>:<value>|<type>|@<sample-rate>|#<tag1-key>:<tag1-value>,<tag2-k/v>`

### Counter

`<name>:<value>|c|@<sample-rate>|#<tag1-key>:<tag1-value>`

It supports sample rate.

### Gauge

`name>:<value>|g|@<sample-rate>|#<tag1-key>:<tag1-value>`

> It does not support sample rate.

Reviewer: ```suggestion
Sample rate is not supported for gauges.
```

## Aggregation

Counters and gauges are aggregated separately.

> The examples below mean receiving metrics in the same interval.

Reviewer: Not sure what this means

Author: Will revise this part.
